scr/main/java/bittercode/util/DatabaseConfig.java

```diff
@@ -0,0 +1,30 @@
+package com.bittercode.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class DatabaseConfig {
+
+    static Properties prop = new Properties();
+    static {
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        InputStream input = classLoader.getResourceAsStream("application.properties");

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider closing InputStream after use</b></div>
<div id="fix">

The `InputStream` is not being closed after use which could lead to resource leaks
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -12,8 +12,7 @@
          ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
 -        InputStream input = classLoader.getResourceAsStream("application.properties");
 -        try {
 +        try (InputStream input = classLoader.getResourceAsStream("application.properties")) {
              prop.load(input);
          } catch (IOException e) {
              e.printStackTrace();
          }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2514333017>#855223</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged